client/src/partials/vouchers/complex.html

```diff
@@ -9,61 +9,90 @@
-                    ng-model="ComplexVoucherCtrl.voucher.currency_id"
-                    ng-value="currency.id"
-                    data-currency-option="{{ currency.id }}"
-                    required>
-                  {{ currency.label }}
-                </label>
+              <!-- right  -->
+              <div class="col-md-6">
+
+                <div class="form-group"
+                  ng-class="{ 'has-error' : ComplexVoucherForm.$submitted && ComplexVoucherForm.type_id.$invalid }">
+                  <label class="control-label">
+                    {{ 'FORM.LABELS.TRANSFER_TYPE' | translate }}
+                    <i uib-popover="{{ 'VOUCHERS.GLOBAL.TRANSFER_TYPE_INFO' | translate }}"
+                      popover-trigger="'mouseenter'"
+                      popover-placement="bottom-left"
+                      class="fa fa-info-circle">
+                    </i>
+                  </label>
+
+                  <!-- financial transaction  -->
+                  <select class="form-control"
+                    name="type_id"
+                    ng-model="ComplexVoucherCtrl.voucher.type_id"
+                    ng-change="ComplexVoucherCtrl.buildDescription()"
+                    ng-disabled="!ComplexVoucherCtrl.financialTransaction" ng-required="ComplexVoucherCtrl.financialTransaction">
+                    <option value="" disabled>--- {{ 'FORM.SELECT.TRANSFER_TYPE' | translate }} ---</option>
 
-                <div class="help-block" ng-messages="ComplexVoucherForm.currency.$error" ng-show="ComplexVoucherForm.$submitted">
-                  <div ng-messages-include="partials/templates/messages.tmpl.html"></div>
+                    <!-- incomes  -->
+                    <optgroup label="{{ 'VOUCHERS.SIMPLE.INCOME' | translate }}">
+                      <option value="{{ item }}" data-item="{{ item.id }}" ng-repeat="item in ComplexVoucherCtrl.incomes">{{ item.text | translate }}</option>
+                    </optgroup>
+
+                    <!-- expenses  -->
+                    <optgroup label="{{ 'VOUCHERS.SIMPLE.EXPENSE' | translate }}">
+                      <option value="{{ item }}" data-item="{{ item.id }}" ng-repeat="item in ComplexVoucherCtrl.expenses">{{ item.text | translate }}</option>
+                    </optgroup>
+                  </select>
+                  <p ng-show="ComplexVoucherCtrl.financialTransaction && !ComplexVoucherForm.$submitted" class="text-info">
+                    <i class="fa fa-info-circle"></i> {{ 'VOUCHERS.GLOBAL.USE_FINANCIAL_ACCOUNT' | translate }}
+                  </p>
+                  <div class="help-block" ng-messages="ComplexVoucherForm.type_id.$error" ng-show="ComplexVoucherForm.$submitted">
+                    <div ng-messages-include="partials/templates/messages.tmpl.html"></div>
+                  </div>
                 </div>
+
+                <bh-currency-select
+                  currency-id="ComplexVoucherCtrl.voucher.currency_id">
```

Comment:
This component needs a `validationTrigger` so that it correctly validates on form submission.